+nosnoc/+solver/MpccSolver.m

```diff
@@ -72,16 +69,294 @@
+                            sigma = nlp.p.sigma_p(); 
+                          case HomotopySteeringStrategy.ELL_INF
+                            sigma = nlp.w.s_elastic(); % Here s_elastic takes the role of sigma in direct, and sigma_p is used to define a penalty parameter for the elastic variable s_elastic
+                          case HomotopySteeringStrategy.ELL_1
+                            % adding elastic variables to nlp.w which augments the original mpcc.w
+                            % Remark: ELL_1 with s_elastic is equivalent to usually Ell_1 penality approach, but this indirect way helps
+                            % to add some constraint on s_elastic (which  avoids unbounded problems somtimes, and it can also improve convergence)
+                            nlp.w.(s_elastic_name) = {{s_elastic_name, size(G_curr, 1)}, opts.s_elastic_min, opts.s_elastic_max, opts.s_elastic_0};
+                            if opts.decreasing_s_elastic_upper_bound
+                                % TODO(@anton) this creates a performance bottleneck
+                                nlp.g.([s_elastic_name '_ub']) = {nlp.w.(s_elastic_name)() - nlp.p.sigma_p(), -inf, 0};
+                            end
+
+                            % TODO(@anton) this creates a performance bottleneck
+                            sigma = nlp.w.(s_elastic_name)();
+                            sum_elastic = sum_elastic + sum1(sigma);
+                        end
+
+                        [ind_scalar_G,ind_nonscalar_G, ind_map_G] = find_nonscalar(G_curr, nlp.w.sym);
+                        [ind_scalar_H,ind_nonscalar_H, ind_map_H] = find_nonscalar(H_curr, nlp.w.sym);
+
+                        obj.ind_map_G = [obj.ind_map_G, ind_map_G];
+                        obj.ind_map_H = [obj.ind_map_H, ind_map_H];
+                        
+                        if opts.lift_complementarities
+                            nlp.w.([name '_G_lift']) = {{'G', length(ind_nonscalar_G)}, 0, inf};
+                            G_lift = G_curr(ind_nonscalar_G);
+                            G_curr(ind_nonscalar_G) = nlp.w.([name '_G_lift'])();
+
+                            nlp.w.([name '_H_lift']) = {{'H', length(ind_nonscalar_H)}, 0, inf};
+                            H_lift = H_curr(ind_nonscalar_H);
+                            H_curr(ind_nonscalar_H) = nlp.w.([name '_H_lift'])();
+                            
+                            nlp.g.([name '_G_lift']) = {nlp.w.([name '_G_lift'])()-G_lift};
+                            nlp.g.([name '_H_lift']) = {nlp.w.([name '_H_lift'])()-H_lift};    
+                        end
+
+                        if ~opts.assume_lower_bounds % Lower bounds on G, H, not already present in MPCC
+                            if ~opts.lift_complementarities
+                                if ~isempty(ind_nonscalar_G)
+                                    nlp.g.([name '_G_lb']) = {G_curr(ind_nonscalar_G), 0, inf};
```

Comment:
one could also have a bigM upper bound, to prevent "blowing up" of the comp variables, altough these bounds are present elsewhere

Reply:
This could be a setting.  Though for direct homotopy this is not a huge issue, and for others `decreasing_s_elastic_upper_bound` is effectively equivalent.